File: examples/Yield.py
#!/usr/bin/env python3

# pylint: disable = bad-whitespace
# pylint: disable = eval-used
# pylint: disable = invalid-name
# pylint: disable = missing-docstring

# --------
# Yield.py
# --------

def f () :
    yield 2
    yield 3
    yield 4

def test1 () :
    p = f()
    assert p is iter(p)
    n = next(p)
    assert n == 2
    n = next(p)
    assert n == 3
    n = next(p)
    assert n == 4
    try :
        n = next(p)
    except StopIteration :
        pass

def test2 () :
    p = f()
    assert list(p) == [2, 3, 4]
    assert list(p) == []

def g () :
    for v in [2, 3, 4] :
        yield v

def test3 () :
    p = g()
    assert p is iter(p)
    n = next(p)
    assert n == 2
    n = next(p)
    assert n == 3
    n = next(p)
    assert n == 4
    try :
        n = next(p)
    except StopIteration :
        pass

def test4 () :
    p = g()
    assert list(p) == [2, 3, 4]
    assert list(p) == []

def main () :
    print("Yield.py")
    for i in range(4) :
        eval("test" + str(i + 1) + "()")
    print("Done.")

if __name__ == "__main__" : # pragma: no cover
    main()

File: examples/Yield.js
#!/usr/bin/env node

/* jshint
    esversion: 6,
    evil:      true
*/

// --------
// Yield.js
// --------

"use strict";

const assert = require('assert');
const _      = require('lodash');

function* f () {
    yield 2;
    yield 3;
    yield 4;}

function test1 () {
    const p = f();
    assert(p === p[Symbol.iterator]());
    var n = p.next();
    assert(!n.done);
    assert(n.value === 2);
    n = p.next();
    assert(!n.done);
    assert(n.value === 3);
    n = p.next();
    assert(!n.done);
    assert(n.value === 4);
    n = p.next();
    assert(n.done);}

function test2 () {
    const p = f();
    const a = [];
    for (let v of p)
        a.push(v);
    assert(_.isEqual(a, [2, 3, 4]));}

function* g () {
    for (let v of [2, 3, 4])
        yield v;}

function test3 () {
    const p = g();
    assert(p === p[Symbol.iterator]());
    var n = p.next();
    assert(!n.done);
    assert(n.value === 2);
    n = p.next();
    assert(!n.done);
    assert(n.value === 3);
    n = p.next();
    assert(!n.done);
    assert(n.value === 4);
    n = p.next();
    assert(n.done);}

function test4 () {
    const p = g();
    const a = [];
    for (let v of p)
        a.push(v);
    assert(_.isEqual(a, [2, 3, 4]));}

function main () {
    console.log("Yield.js");
    for (let i of _.range(4))
        eval("test" + (i + 1) + "()");}
    console.log("Done.");

main();
